### DISCUSSION TITLE
focusAtom on another prism focused atom

### DISCUSSION THREAD

**Discussion by quangbuule** [pmndrs/jotai#1035] [Q&A]
In this example, `contentAtom` declaration has a TS problem since `fooAtom` is a result of a prism so its `WritableAtom<Foo | undefined, ...>`

I would love to how we can solve this problem.

https://codesandbox.io/s/clever-dhawan-4bm4nh?file=/src/App.tsx

```ts
import { atom, useAtom } from "jotai";
import { focusAtom } from "jotai/optics";

type Foo = {
  id: string;
  content: string;
};

const foosAtom = atom<Foo[]>([{ id: "1", content: "" }]);
const fooAtom = focusAtom(foosAtom, (optic) =>
  optic.find((one) => one.id === "1")
);

const contentAtom = focusAtom(fooAtom, optic => optic.prop('content')); // TS is complaining sadly

export default function App() {
  const [content] = useAtom(contentAtom);
  return <>{content}</>;
}
```

**Comment by dai-shi** [comment]
In the case of
```ts
const fooAtom = focusAtom(foosAtom, (optic) =>
  optic.find((one) => one.id === "2")
);
```
the type `| undefined` is correct. So, TypeScript can't know if it can remove `| undefined`.

Only type assertion can solve it??
```ts
const fooAtom = focusAtom(foosAtom, (optic) =>
  optic.find((one) => one.id === "1")
) as PrimitiveAtom<Foo>;
```

Not sure if we can have type assertion at optics. Would be nice if it could.

**Reply by quangbuule** [reply]
Yea, the problem is either `focusAtom` or `optics` can't work on optional types like this, which makes it hard to chain atoms.

**Reply by dai-shi** [reply]
I see. If you have chance, please have a look at `focusAtom` source code. I'm not 100% sure if it's correct, so any suggestion might help.
